extensions/amp-analytics/0.1/vendors/parsely.js

```diff
@@ -17,20 +17,28 @@
 import {jsonLiteral} from '../../../../src/json';
 
 const PARSELY_CONFIG = jsonLiteral({
+  'vars': {
+    'data': '{}',
+    'plid': Math.floor(Math.random() * 99999999),
+    'pvid': Math.floor(Math.random() * 99999999),
```

Comment:
From the looks of things, this literal and `Math.random` is only called once which isn't what we want for `plid` or `pvid`. Might want to consult with the AMP team on this or just strip this bit and only introduce `extra_data`.

Reply:
That is actually what we want, I think, because we want those values to persist across all other heartbeat events. I was also thinking that we don't need to worry about supporting dynamic pageviews in the AMP context.

Here's the relevant generation function in the tracker: https://github.com/Parsely/parsely-js/blob/352ea7d554d821ddc75a079022b261f871ab232a/js-api/src/tracker/pixel.js#L142-L147

Reply:
I just built the project and went [here](http://localhost:8000/examples/analytics-vendors.amp.html) and across hard reloads and even an incognito window the pvid and the plid are consistently the same. I think it's computing that file at build time and not runtime. That's why all of the "variables" in it are using the `${variable_name}` jinja-style syntax.

Reply:
ah @cwisecarver that makes sense - when running it locally, it behaved as suspected (new values on every pageload), but that was just using the uncompiled source.

Reply:
In the past, I've done computations in the javascript side and exposed those values as variables and then they're computed at run time. They may already have a "random number  variable" that you could repurpose. Yep, [here's the list](https://github.com/Parsely/amphtml/blob/master/spec/amp-var-substitutions.md), towards the bottom there's a pv id and a "random number".

Reply:
The learning curve on this codebase is just a never ending 45° angle line.

Reply:
we could also use the ${random} variable: https://github.com/ampproject/amphtml/blob/master/spec/amp-var-substitutions.md#random

However, it returns a value between 0 and 1, and I'm not sure how well that will play with our other downstream expectations.

Reply:
Page view id is at least an integer. You could track down the implementation and see if it's okay.

Reply:
After about 15 layers of indirection, the implementation of page view id. https://github.com/ampproject/amphtml/blob/master/src/service/document-info-impl.js#L131-L133

Reply:
Thanks @cwisecarver for that find - sounds like that will be close enough for our purposes.